cmd/buildctl/trace.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"context"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/moby/buildkit/util/appcontext"
+	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+	"github.com/opentracing/opentracing-go/log"
+	jaeger "github.com/uber/jaeger-client-go"
+	"github.com/urfave/cli"
+)
+
+func getTracer() (opentracing.Tracer, io.Closer) {
+	if traceAddr := os.Getenv("JAEGER_TRACE"); traceAddr != "" {
```

Comment:
Why this is not a CLI flag?

Reply:
I don't expect users to type it for specific commands but rather opt-in to tracing and then all commands would be traced on the background. I think it best makes sense as a config option but we haven't implemented that yet. Also, I'm not sure how this would eventually look like. Using Jaeger is definitely not set in stone and there are only ~5 lines that describe the Jaeger specifics. Main thing is to use the opentracing tools for collection.